### PR TITLE
feat(j-s): Reviewer completes indictment review

### PR DESCRIPTION
**Pull request by unakb**
[Ríksak - Saksóknari - Yfirlitssíða fyrir yfirlestur máls](https://app.asana.com/0/1199153462262248/1207159654122357)

## What

Allow a public prosecutor to make a review decision in indictment cases

## Screenshots / Gifs

https://github.com/island-is/island.is/assets/4820859/1dbf56da-ab11-41f5-a50d-bd1fa3314558

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Added a new field `indictmentReviewDecision` to case updates for managing indictment cases.
  - Introduced a review decision component for public prosecutors.
  - Included UI components and internationalized strings for reviewing decisions.

- **Bug Fixes**
  - Updated access permissions for prosecution users to improve case handling.

- **Style**
  - Implemented new CSS styles for the review decision grid layout.

- **Documentation**
  - Enhanced clarity and internationalization in the indictment overview with updated strings.

- **Database Changes**
  - Executed a database migration to support indictment review decisions.

### CODE REVIEW

File: apps/judicial-system/web/src/routes/PublicProsecutor/Indictments/ReviewDecision/ReviewDecision.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    if (!indictmentReviewDecision) {
+      return
+    }
+    const updateSuccess = await updateCase(workingCase.id, {
+      indictmentReviewDecision: indictmentReviewDecision,
+    })
+    if (updateSuccess) {
+      router.push(constants.CASES_ROUTE)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/web/src/routes/PublicProsecutor/Indictments/ReviewDecision/ReviewDecision.tsx:57]: Good point, we have discussed this as a team and this is already the case in plenty of other places in our code so will be kept like this until some later date if we decide to refactor this

File: apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts [hunk 2]
```diff
@@ -44,7 +44,10 @@ const getProsecutionUserCasesQueryFilter = (user: User): WhereOptions => {
       [Op.or]: [
         { prosecutors_office_id: user.institution?.id },
         { shared_with_prosecutors_office_id: user.institution?.id },
-        { indictment_reviewer_id: user.id },
+        [
+          { indictment_reviewer_id: user.id },
+          { indictment_review_decision: null },
+        ],
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts:50]: I agree with the rabbit here.

File: apps/judicial-system/backend/src/app/modules/case/filters/test/cases.filter.spec.ts [hunk 3]
```diff
@@ -55,11 +55,15 @@ describe('getCasesQueryFilter', () => {
...
+
         {
           [Op.or]: [
             { prosecutors_office_id: 'Prosecutors Office Id' },
             { shared_with_prosecutors_office_id: 'Prosecutors Office Id' },
-            { indictment_reviewer_id: 'Prosecutor Id' },
+            [
+              { indictment_reviewer_id: 'Prosecutor Id' },
+              { indictment_review_decision: null },
+            ],
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/filters/test/cases.filter.spec.ts:66]: The tests need to be fixed once the filter has been fixed.

File: apps/judicial-system/backend/src/app/modules/case/filters/test/cases.filter.spec.ts [hunk 4]
```diff
@@ -117,7 +121,10 @@ describe('getCasesQueryFilter', () => {
...
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/case/filters/test/cases.filter.spec.ts:127]: The tests need to be fixed once the filter has been fixed.

**Review by gudjong — CHANGES_REQUESTED** [review]
One issue with case filters.

File: apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (unakb) [reply apps/judicial-system/backend/src/app/modules/case/filters/cases.filter.ts]: Interestingly this way does work as well, but Ive changed it accordingly

**Review by gudjong — APPROVED** [review]

**Pull request merged**